### PR TITLE
add codecov.io

### PR DESCRIPTION
**Pull request by kmaehashi**

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -36,4 +36,10 @@ fi
 pytest_opts+=(-m "${pytest_marks[*]}")
 
 python -m pytest "${pytest_opts[@]}" tests
+
+# Submit coverage to Coveralls
 python ../push_coveralls.py
+
+# Submit coverage to Codecov
+rm -f coverage.xml
```
Comment (niboshi) [inline test.sh]: Why is this needed?

Comment (kmaehashi) [reply test.sh]: Because `codecov` wrongly reuses the previous `coverage.xml` if it is left.

* `codecov` command uses `coverage.xml` to submit coverage.
* `codecov` runs `coverage xml` (which generates `coverage.xml` from `.coverage`) if `coverage.xml` does not exist. https://github.com/codecov/codecov-python/blob/v2.0.15/codecov/__init__.py#L675

Comment (niboshi) [reply test.sh]: If it's cleanup of coveralls, isn't it better to put this line just after coveralls?

Comment (kmaehashi) [reply test.sh]: It is not related to coveralls.

* `.coverage` is a raw coverage data generated by `coverage` package (called via `pytest --cov`). Coveralls requires this file.
* `coverage.xml` is a XML representation of the coverage data (can be converted from `.coverage` by running `coverage xml` command). Codecov requires this file.

Maybe it's better to run `coverage xml` command by ourselves, rather than letting `codecov` command do it?

Comment (niboshi) [reply test.sh]: Ah, do you mean it's cleanup of the previous codecov run if the test is run consecutively?
So, leaving a comment like "cleanup the remnant from the previous test run" would suffice.

Comment (kmaehashi) [reply test.sh]: That's right, sorry for making confusion.

Anyway I fixed to explicitly generate `coverage.xml` so that we don't have to care about the trash from previous codecov run.

**Comment by niboshi** [comment]
LGTM.

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
@kmaehashi 
Jenkins test fails every time with the same error.
Is it relevant to this PR?

```
TestToGPUScalar_param_8.test_numpy_scalar
```

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
Test has passed.
LGTM!
